### ISSUE TITLE
Found unparsable section when using keyword as column name

### ISSUE THREAD

**Issue by ossinkine** [sqlfluff/sqlfluff#3412]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When using reserved keyword as column name the parsing error occured but this is correct syntax.

### Expected Behaviour

Parsing successful and linter error if using keyword as column name prohibided

### Observed Behaviour

Parsing error occured

### How to reproduce

```
$ echo 'select id, start, end from periods' | sqlfluff lint - --dialect=postgres
== [stdin] FAIL
L:   1 | P:   1 | L036 | Select targets should be on a new line unless there is
                       | only one select target.
L:   1 | P:  19 |  PRS | Line 1, Position 19: Found unparsable section: 'end'
All Finished 📜 🎉!
```

### Dialect

`postgres`

### Version

```
$ sqlfluff --version
sqlfluff, version 0.13.2
```
```
$ python --version
Python 3.9.13
```

### Configuration

Default configuration, no configuration files

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
According this this: https://www.postgresql.org/docs/current/sql-keywords-appendix.html `END` is a reserved word so doesn’t that mean it can’t be used as a column name?

**Comment by ossinkine** [comment]
@tunetheweb You're right, Postgres doesn't allow to use it as column name, but allow to use with reference
```sql
select id, start, periods.end from periods
```
This query is correct for Postgres but unparsable for SQLFluff:
```
$ echo 'select id, start, periods.end from periods' | sqlfluff lint - --dialect=postgres
== [stdin] FAIL
L:   1 | P:   1 | L036 | Select targets should be on a new line unless there is
                       | only one select target.
L:   1 | P:  26 |  PRS | Line 1, Position 26: Found unparsable section: '.end'
All Finished 📜 🎉!
```

**Comment by tunetheweb** [comment]
Sorry I don’t get what you mean by “but allow to use with reference”?

Do you mean you can’t create a column named that (without using quotes), but if you do that (with quotes) then you can reference it without quotes?

**Comment by ossinkine** [comment]
I mean it's allowed to use keyword as column name without quotes if it's prefixed by table name, see query example above

**Comment by tunetheweb** [comment]
So your first query in the initially comment was incorrect and should have caused a parse error, but the one you added in the subsequent comment is valid in Postgres?

I still think that sounds like pretty bad practice to be honest and would suggest quoting it there too for consistency (I presume you can quote it if you want?). The job of the linter is to highlight bad practices so, although SQLFluff is highlighting this in the wrong way (as a parse error, rather than a rule lint error) I still think it’s highlighting a bad practice.

Would using quotes be an option for you?

**Comment by ossinkine** [comment]
@tunetheweb 

> So your first query in the initially comment was incorrect and should have caused a parse error, but the one you added in the subsequent comment is valid in Postgres?

Correct

> I still think that sounds like pretty bad practice to be honest and would suggest quoting it there too for consistency (I presume you can quote it if you want?). The job of the linter is to highlight bad practices so, although SQLFluff is highlighting this in the wrong way (as a parse error, rather than a rule lint error) I still think it’s highlighting a bad practice.
> 
> Would using quotes be an option for you?

As workaround I use quotes but would be better to get lint error instead of parsing error to be able configure it or ignore.
